packages/xml-views-validation/src/validators/elements/cardinality-of-aggregation.ts

```diff
@@ -0,0 +1,87 @@
+import { filter, forEach, sortBy, map } from "lodash";
+import { XMLElement } from "@xml-tools/ast";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { getUI5AggregationByXMLElement } from "@ui5-language-assistant/logic-utils";
+import { CardinalityOfAggregationIssue } from "../../../api";
+
+export function validateCardinality(
+  xmlElement: XMLElement,
+  model: UI5SemanticModel
+): CardinalityOfAggregationIssue[] {
+  const aggregation = getUI5AggregationByXMLElement(xmlElement, model);
+  if (aggregation === undefined || xmlElement.syntax.openName === undefined) {
+    return [];
+  }
+
+  const allCousinsElements = getAllICousinsElements(xmlElement);
+  if (allCousinsElements.length > 1 && aggregation.cardinality === "0..1") {
+    const firstCousin = getFirstCousin(allCousinsElements);
+    // The error will not be shown on first cousin element
+    const currentAggregationSubElements = filter(
+      xmlElement.subElements,
+      (_) => !isFirstCousin(_, firstCousin)
+    );
+
+    const cardinalityOfAggregationIssues = map(
+      currentAggregationSubElements,
+      (_): CardinalityOfAggregationIssue => ({
+        kind: "CardinalityOfAggregation",
+        message: `The aggregation '${xmlElement.name}' has cardinality of 0..1 and can only contain one element.`,
+        severity: "error",
+        offsetRange: {
+          start: (_.syntax.openName ?? _.position).startOffset,
```

Comment:
What is the purpose of this  condition `(_.syntax.openName ?? _.position)` here?